packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -59,15 +69,49 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
     span.setAttribute('http.method', startContext.method)
     span.setAttribute('http.url', networkRequestInfo.url)
 
-    return (endContext: RequestEndContext) => {
-      if (endContext.state === 'success') {
-        span.setAttribute('http.status_code', endContext.status)
-        this.spanFactory.endSpan(span, endContext.endTime)
-      }
+    return {
+      onRequestEnd: (endContext: RequestEndContext) => {
+        if (endContext.state === 'success') {
+          span.setAttribute('http.status_code', endContext.status)
+          this.spanFactory.endSpan(span, endContext.endTime)
+        }
+      },
+      extraRequestHeaders: this.getExtraRequestHeaders(
+        startContext,
+        span
+      )
     }
   }
 
   private shouldTrackRequest (startContext: RequestStartContext): boolean {
     return startContext.url !== this.configEndpoint && permittedPrefixes.some((prefix) => startContext.url.startsWith(prefix))
   }
+
+  private getExtraRequestHeaders (startContext: RequestStartContext, span?: SpanInternal): Record<string, string> {
+    const extraRequestHeaders: Record<string, string> = {}
+
+    if (this.tracePropagationUrls.some(regexp => regexp.test(startContext.url))) {
+      if (span && span.parentSpanId) {
+        const traceId = span.traceId
+        const parentSpanId = span.parentSpanId
+        const sampled = span.samplingRate <= this.spanFactory.sampler.spanProbability.scaled
```

Comment:
Would this logic be better off in the sampler?

Reply:
which logic, this?

```
const sampled = currentSpanContext.samplingRate <= this.spanFactory.sampler.spanProbability.scaled
```

e.g. like this?

```
const sampled = this.spanFactory.sampler(currentSpanContext.samplingRate)
```

Reply:
I was more just thinking of a method on `sampler` called `sample` or `shouldSample`.  Just to have that logic (i.e. the `<=` test in a single place.

Reply:
```
const sampled = this.spanFactory.sampler(currentSpanContext.samplingRate)
```

Should have been

```
const sampled = this.spanFactory.sampler.shouldSample(currentSpanContext.samplingRate)
```

So it would be used like this in the two places:

```
const sampled = this.spanFactory.sampler.shouldSample(currentSpanContext.samplingRate)
// or, if a span is being created
const sampled = this.spanFactory.sampler.shouldSample(span.samplingRate)
```